scene/gui/foldable_container.cpp

```diff
@@ -0,0 +1,301 @@
+		set_expanded(!expanded);
+		accept_event();
+		return;
+	}
+
+	Ref<InputEventMouseButton> b = p_event;
+
+	if (b.is_valid()) {
+		if (b->is_pressed() && b->get_button_index() == MouseButton::LEFT && b->get_position().y <= title_size.height) {
+			set_expanded(!expanded);
+			accept_event();
+			return;
+		}
+	}
+}
+
+void FoldableContainer::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_DRAW: {
+			RID ci = get_canvas_item();
+
+			Ref<Texture2D> icon = expanded ? theme_cache.arrow : theme_cache.arrow_collapsed;
+			Ref<StyleBox> title_style;
+			if (is_hovering) {
+				title_style = expanded ? theme_cache.title_hover_style : theme_cache.title_collapsed_hover_style;
+			} else {
+				title_style = expanded ? theme_cache.title_style : theme_cache.title_collapsed_style;
+			}
+
+			if (title_style.is_valid()) {
+				title_size = Point2(get_size().width, title_style->get_minimum_size().height);
+			}
+			if (theme_cache.title_font.is_valid()) {
+				title_size.height += theme_cache.title_font->get_height(theme_cache.title_font_size);
+			}
+			if (icon.is_valid()) {
+				if (title_style.is_valid()) {
+					title_size.height = MAX(title_size.height, title_style->get_minimum_size().height + icon->get_height());
+				} else {
+					title_size.height = MAX(title_size.height, icon->get_height());
+				}
+			}
+
+			Point2 title_pos;
+			int title_width = 0;
+			if (title_style.is_valid()) {
+				title_style->draw(ci, Rect2(Point2(), title_size));
+				set_custom_minimum_size(Size2(0, title_size.height));
+				title_pos = title_style->get_offset();
+				title_width = get_size().width - title_style->get_minimum_size().width;
+			}
+
+			if (theme_cache.title_font.is_valid()) {
+				title_pos.y += theme_cache.title_font->get_ascent(theme_cache.title_font_size);
+			}
+
+			if (icon.is_valid()) {
+				int h_separation = CLAMP(theme_cache.h_separation, 0, title_width - icon->get_width() - 1);
+				title_pos.x += icon->get_width() + h_separation;
+				title_width -= icon->get_width() + h_separation;
+				Point2 icon_pos;
+				if (title_style.is_valid()) {
+					icon_pos = title_style->get_offset();
+				}
+				if (theme_cache.title_font.is_valid()) {
+					if (theme_cache.title_font->get_height(theme_cache.title_font_size) <= icon->get_height()) {
+						title_pos.y += (icon->get_height() - theme_cache.title_font->get_ascent(theme_cache.title_font_size)) / 2;
+					} else {
+						icon_pos.y += (theme_cache.title_font->get_height(theme_cache.title_font_size) - icon->get_height()) / 2;
+					}
+				}
+				icon->draw(ci, icon_pos);
+			}
+
+			if (theme_cache.title_font.is_valid()) {
+				Color font_color = expanded ? theme_cache.title_font_color : theme_cache.title_collapsed_font_color;
+				if (is_hovering) {
+					font_color = theme_cache.title_hover_font_color;
+				}
+				draw_string(theme_cache.title_font, title_pos, title, title_alignment, title_width, theme_cache.title_font_size, font_color);
```

Comment:
`draw_string()` is used temporarily, it should be replaced later after adding RTL text orientation and translation.